specs/protocol/isthmus/exec-engine.md

```diff
@@ -63,6 +79,33 @@ places a burden on users of the system in a post-fault-proofs world, where:
 Placing the [`L2ToL1MessagePasser`][l2-to-l1-mp] account storage root in the `withdrawalsRoot` field alleviates this burden
 for users and protocol participants alike, allowing them to propose and verify other proposals with lower operating costs.
 
+#### Genesis Block
+
+If isthumus is active at the genesis block, the withdrawals root is the empty withdrawals root, regardless of L2 state.
```

Comment:
It's not so much that there's utility in proposing for the genesis block, but that this breaks two invariants:
1. That starting from Isthmus the withdrawalsRoot header is the `L2ToL1MessagePasser` storage root
2. That the safe head can always be used to produce a valid proposal

Everywhere that reads the withdrawalRoot would need to have special case handling for genesis. So I'd be very inclined to have the extra state read in L2 genesis generation to make all other cases simple.

Reply:
As an example of the bugs that come from this special case - this PR adds a paragraph to `proposal.md` that says the withdrawalsRoot can be used post-Isthmus but forgets the genesis special case where it can't.

Reply:
The withdrawal-contract genesis storage-root will keep changing, and having many special genesis values to put in the block-header makes things complicated. It couples the state content and block-header generation to tightly. We should just default the withdrawals-root in the genesis block to an empty withdrawals MPT root. I think not being allowed to withdraw from the genesis block is fine, as there aren't supposed to be withdrawals anyway.

Reply:
Typically to create genesis block you populate the world state from the genesis config and then insert that into the stateRoot field of the block header. That already includes the withdrawal contract storage root so there doesn't seem to be any additional dependencies being added here.

My concern is less about whether you can withdraw but more that it's really easy to forget genesis is a special case and do things like make invalid proposals or depend on the withdrawalsRoot as mentioned above when it's not actually accurate. Given we already have to know the genesis world state to create the block header it doesn't seem worth creating that foot gun.

Reply:
I met with @protolambda/ @ajsutton and we came to the agreement that even for the genesis block, we will set the `withdrawalsRoot` field in the header to the `L2ToL1MessagePasser` storage root, if Isthmus is active -- this is a change from the original spec proposal that stated that it will be set to the empty withdrawals root.